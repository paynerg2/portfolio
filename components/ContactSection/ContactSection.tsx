import { Box, Heading, HStack, Stack, Text, Divider, Icon, IconButton } from '@chakra-ui/react';
import { sectionNames } from '../../utils/constants';
import ContactForm from '../ContactForm';
import { MdOutlineMailOutline, MdPhoneAndroid } from 'react-icons/md';
import { FaGithub, FaLinkedin } from 'react-icons/fa';

const ContactSection = () => {
    const contrastingTextColor = 'blue.300';
    const headingStyles = {
        size: 'md',
    };

    return (
        <Stack direction="column" gap={10} alignItems="flex-start">
            <Box>
                <Heading id={sectionNames.contact}>Contact Me</Heading>
                <Divider orientation="horizontal" w={100} ml={10} mb={5} h={1} />
                <Text color={contrastingTextColor}>
                    Have any questions, or want to work with me? Send me a message!
                </Text>
            </Box>
            <Stack direction="row" spacing={10} justify="center" w="100%">
                <ContactForm />
                <Stack
                    spacing={5}
                    display="flex"
                    flexGrow={1}
                    direction="column"
                    w="25%"
                    align="center"
                >
                    <Stack direction="row" align="center" gap={1}>
                        <Icon as={MdOutlineMailOutline} />
                        <Heading {...headingStyles}>Email</Heading>
                    </Stack>
                    <Text color={contrastingTextColor}>paynerg2@gmail.com</Text>
                    <Stack direction="row" align="center" gap={1}>
                        <Icon as={MdPhoneAndroid} />
                        <Heading {...headingStyles}>Phone</Heading>
                    </Stack>
                    <Text color={contrastingTextColor}>(919)-428-0993</Text>
                    <HStack spacing={4}>
                        <IconButton
                            as="a"
                            href="https://github.com/paynerg2"
                            target="_blank"
                            rel="noopener noreferrer"
                            icon={<FaGithub />}
                            aria-label="Github"
                        />
                        <IconButton
                            as="a"
                            href="https://linkedin.com/in/paynerg2"
                            target="_blank"
                            rel="noopener noreferrer"
                            icon={<FaLinkedin />}
                            aria-label="LinkedIn"
                        />
                    </HStack>
                </Stack>
            </Stack>
        </Stack>
    );
};

export default ContactSection;
