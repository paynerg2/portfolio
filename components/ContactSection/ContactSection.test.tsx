import { render } from '@testing-library/react';
import ContactSection from './ContactSection';

describe('Contact Section', () => {
    it('renders the component', () => {
        const component = render(<ContactSection />);
        expect(component).toMatchSnapshot();
    });
});
