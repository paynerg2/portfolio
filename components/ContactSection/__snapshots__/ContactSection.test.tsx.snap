// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Section renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-134wq2"
      >
        <div
          class="css-0"
        >
          <h2
            class="chakra-heading css-0"
            id="contact"
          >
            Contact Me
          </h2>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-g3647t"
          />
          <p
            class="chakra-text css-z2nxer"
          >
            Have any questions, or want to work with me? Send me a message!
          </p>
        </div>
        <div
          class="chakra-stack css-8l053e"
        >
          <form
            style="width: 100%;"
          >
            <div
              class="chakra-stack css-137fd6w"
            >
              <div
                class="chakra-stack css-vy21t6"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <input
                    class="chakra-input css-o9hh3a"
                    id="name"
                    name="name"
                    placeholder="Name"
                  />
                </div>
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <input
                    class="chakra-input css-o9hh3a"
                    id="email"
                    name="email"
                    placeholder="Email"
                    type="email"
                  />
                </div>
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <input
                  class="chakra-input css-o9hh3a"
                  id="subject"
                  name="subject"
                  placeholder="Subject"
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <textarea
                  class="chakra-textarea css-1gnsnpc"
                  id="message"
                  name="message"
                  placeholder="Message"
                />
              </div>
              <button
                class="chakra-button css-nxxp8o"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
          <div
            class="chakra-stack css-1rzwfrg"
          >
            <div
              class="chakra-stack css-1a4j7nd"
            >
              <svg
                class="chakra-icon css-13otjrl"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                />
              </svg>
              <h2
                class="chakra-heading css-0"
              >
                Email
              </h2>
            </div>
            <p
              class="chakra-text css-z2nxer"
            >
              paynerg2@gmail.com
            </p>
            <div
              class="chakra-stack css-1a4j7nd"
            >
              <svg
                class="chakra-icon css-13otjrl"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
                />
              </svg>
              <h2
                class="chakra-heading css-0"
              >
                Phone
              </h2>
            </div>
            <p
              class="chakra-text css-z2nxer"
            >
              (919)-428-0993
            </p>
            <div
              class="chakra-stack css-1qwdsol"
            >
              <a
                aria-label="Github"
                class="chakra-button css-198bfjk"
                href="https://github.com/paynerg2"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
              <a
                aria-label="LinkedIn"
                class="chakra-button css-198bfjk"
                href="https://linkedin.com/in/paynerg2"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-134wq2"
    >
      <div
        class="css-0"
      >
        <h2
          class="chakra-heading css-0"
          id="contact"
        >
          Contact Me
        </h2>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-g3647t"
        />
        <p
          class="chakra-text css-z2nxer"
        >
          Have any questions, or want to work with me? Send me a message!
        </p>
      </div>
      <div
        class="chakra-stack css-8l053e"
      >
        <form
          style="width: 100%;"
        >
          <div
            class="chakra-stack css-137fd6w"
          >
            <div
              class="chakra-stack css-vy21t6"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <input
                  class="chakra-input css-o9hh3a"
                  id="name"
                  name="name"
                  placeholder="Name"
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <input
                  class="chakra-input css-o9hh3a"
                  id="email"
                  name="email"
                  placeholder="Email"
                  type="email"
                />
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <input
                class="chakra-input css-o9hh3a"
                id="subject"
                name="subject"
                placeholder="Subject"
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <textarea
                class="chakra-textarea css-1gnsnpc"
                id="message"
                name="message"
                placeholder="Message"
              />
            </div>
            <button
              class="chakra-button css-nxxp8o"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
        <div
          class="chakra-stack css-1rzwfrg"
        >
          <div
            class="chakra-stack css-1a4j7nd"
          >
            <svg
              class="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
              />
            </svg>
            <h2
              class="chakra-heading css-0"
            >
              Email
            </h2>
          </div>
          <p
            class="chakra-text css-z2nxer"
          >
            paynerg2@gmail.com
          </p>
          <div
            class="chakra-stack css-1a4j7nd"
          >
            <svg
              class="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
              />
            </svg>
            <h2
              class="chakra-heading css-0"
            >
              Phone
            </h2>
          </div>
          <p
            class="chakra-text css-z2nxer"
          >
            (919)-428-0993
          </p>
          <div
            class="chakra-stack css-1qwdsol"
          >
            <a
              aria-label="Github"
              class="chakra-button css-198bfjk"
              href="https://github.com/paynerg2"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 496 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                />
              </svg>
            </a>
            <a
              aria-label="LinkedIn"
              class="chakra-button css-198bfjk"
              href="https://linkedin.com/in/paynerg2"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
